nan.h

```diff
@@ -801,44 +801,83 @@ class TryCatch {
 #endif
 
   NAN_INLINE v8::Local<v8::Value> MakeCallback(
-      v8::Local<v8::Object> target
+      v8::Local<v8::Value> target
     , v8::Local<v8::Function> func
     , int argc
     , v8::Local<v8::Value>* argv) {
+    v8::Isolate *isolate = v8::Isolate::GetCurrent();
+    v8::EscapableHandleScope scope(isolate);
+    v8::Local<v8::Object> target_obj;
 #if NODE_MODULE_VERSION < IOJS_3_0_MODULE_VERSION
-    return New(node::MakeCallback(
-        v8::Isolate::GetCurrent(), target, func, argc, argv));
+    if (target->IsUndefined() || target->IsNull()) {
+      target_obj = isolate->GetCurrentContext()->Global();
```

Comment:
Doing a straight cast led to a segfault. Can't access `function.caller`, because I don't have access to the current closure from within `Nan::MakeCallback`. Even if I could, it would still not be possible to make target `Undefined`, since `node::MakeCallback` only accepts `Object`.

I did some testing, in strict mode, doing `Function::Call` with `Undefined` as target produced `undefined`, while setting `Null` as target produced `null`. In normal mode, both produce `global`.

All in all, it seems the current behavior is the only one that can be consistently implemented and this PR can be closed. No `Value` targets for anyone.